zipline-security/src/jsMain/kotlin/app/cash/zipline/security/SecureRandom.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2024 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.security
+
+class SecureRandom internal constructor(
```

Comment:
```suggestion
internal class SecureRandom(
```
If the class cannot be externally constructed there's no reason to put it into the public API. Seems like we should _only_ be exposing a single random type that is always secure.

Reply:
I’d like a Random interface to support deterministic testing, and a secure implementation for production use.

Reply:
I agree that a Random interface needs to have a way to be mocked / faked in a deterministic way for testing. Can we not make the js type internal only and still fake the backend implementation?

Reply:
Right so that still necessitates only a single type which is mostly what I'm getting at. There's just one `Random` and you can either use factory functions to create a cryptographically-secure instance, or a seeded fake.